### PR TITLE
thomas python example

### PR DESCRIPTION
**Pull request by gammison**

### CODE REVIEW

File: chapters/matrix_methods/thomas/code/python/thomas.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#Author: gammison
+
+#note this example is inplace and destructive
+def thomas(a:list,b:list,c:list,d:list):
```
Comment (VikingScientist) [inline chapters/matrix_methods/thomas/code/python/thomas.py]: We don't use type hints elsewhere in the python code. The code should have the same style and I would request that this PR leave it out. If you strongly feel that type hints are a good thing, then either make an issue discussing it or create a brand new PR introducing it in all python codes.

File: chapters/matrix_methods/thomas/code/python/thomas.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    
+    #do the back substitution, will loop n-2 to 0 inclusive
+    for i in range(n-2,-1,-1):
+        d[i] = d[i] - c[i]*d[i+1]
+    
+    return d
+    
+    #Complexity:
+    #Two loops,each take O(n), and a few contant time operations => T(n) = 2*O(n)+c*O(1) = O(n)
```
Comment (VikingScientist) [inline chapters/matrix_methods/thomas/code/python/thomas.py]: Complexity discussions should be left to the accompanying article and not in the source code. Please remove this

File: chapters/matrix_methods/thomas/code/python/thomas.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+#                [.8666]
+#soln will equal [1.533]
+#                [-.266]
+#note we index a from 1 and c from 0
+a=[0,2,3]
+b=[1,3,6]
+c=[4,5,0]
+d=[7,5,3]
+soln = thomas(a,b,c,d)
+print(soln)
```
Comment (VikingScientist) [inline chapters/matrix_methods/thomas/code/python/thomas.py]: Wrap execution code into a `main`-method. See i.e. `data_compression/huffman/code/python/huffman.py`

Comment (Butt4cak3) [reply chapters/matrix_methods/thomas/code/python/thomas.py]: [Link for the lazy](https://github.com/algorithm-archivists/algorithm-archive/blob/a9ef5ff179c2cba47fb564cc04043da2c46c03df/chapters/data_compression/huffman/code/python/huffman.py#L93-L115)

Comment (gammison) [reply chapters/matrix_methods/thomas/code/python/thomas.py]: style changes pushed

**Comment by gammison** [comment]
Merge from #188 resolved and style fixes completed

File: chapters/matrix_methods/thomas/code/python/thomas.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#Author: gammison
```
Comment (Butt4cak3) [inline chapters/matrix_methods/thomas/code/python/thomas.py]: All the other Python code in the project puts a space between `#` and the comment text.

File: chapters/matrix_methods/thomas/code/python/thomas.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+#note this example is inplace and destructive
+def thomas(a,b,c,d):
```
Comment (Butt4cak3) [inline chapters/matrix_methods/thomas/code/python/thomas.py]: There should be spaces after every comma in the parameter list.

File: chapters/matrix_methods/thomas/code/python/thomas.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    
+    #set the initial elements
+    c[0] = c[0] / b[0]
+    d[0] = d[0] / b[0]
+
+    n = len(d) #number of equations to solve
+    #range is exclusive, loop will run from 1 to n-1
```
Comment (Butt4cak3) [inline chapters/matrix_methods/thomas/code/python/thomas.py]: I think `range()` is common in enough to Python to leave this comment out.

File: chapters/matrix_methods/thomas/code/python/thomas.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    #[0  3  6][z]   [3]
+    #                [.8666]
+    #soln will equal [1.533]
+    #                [-.266]
+    #note we index a from 1 and c from 0
+    a=[0,2,3]
+    b=[1,3,6]
+    c=[4,5,0]
+    d=[7,5,3]
+    soln = thomas(a,b,c,d)
```
Comment (Butt4cak3) [inline chapters/matrix_methods/thomas/code/python/thomas.py]: Again, add some spaces before and after `=` and after every `,` to be in line with all the other code examples.

Comment (gammison) [reply chapters/matrix_methods/thomas/code/python/thomas.py]: Done

**Comment by VikingScientist** [comment]
This looks good. I have no more objections and I think it's ready for merging

**Comment by Butt4cak3** [comment]
I agree.

**Pull request merged**